message_sender/serializers.py

```diff
@@ -21,9 +21,10 @@ class Meta:
 
     def to_internal_value(self, data):
         if data.get('channel'):
-            channel_id = data.pop('channel')
-            data['channel_channel_id'] = channel_id
+            self.fields['channel'] = serializers.PrimaryKeyRelatedField(
+                 queryset=Channel.objects.all())
```

Comment:
I think it would be better if we put the `PrimaryKeyRelatedField` on the class of the serialiser, rather than here. That should also remove the need for the channel existing below (I think)